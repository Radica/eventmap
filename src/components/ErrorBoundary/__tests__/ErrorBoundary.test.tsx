import React, { ReactNode } from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';

import ErrorBoundary from '../index';

describe('<ErrorBoundary />', () => {
    const tree = (children?: ReactNode) =>
        render(
            <MemoryRouter>
                <ErrorBoundary>{children}</ErrorBoundary>
            </MemoryRouter>
        ).container.firstChild;

    it('renders nothing if no children', () => {
        expect(tree()).toMatchSnapshot();
    });

    it('renders children if no error', () => {
        expect(
            tree(
                <div>
                    <h1>I am Welly</h1>
                </div>
            )
        ).toMatchSnapshot();
    });

    it('renders error view if an error occurs', () => {
        global.console.error = jest.fn();

        tree(<div>{new Error()}</div>);

        expect(screen.getByTestId('error-view')).toBeInTheDocument();
    });
});
