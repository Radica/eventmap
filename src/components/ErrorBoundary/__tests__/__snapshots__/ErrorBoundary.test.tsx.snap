// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ErrorBoundary /> renders children if no error 1`] = `
<div>
  <h1>
    I am Welly
  </h1>
</div>
`;

exports[`<ErrorBoundary /> renders nothing if no children 1`] = `null`;
